cogs/autoemoji.py

```diff
@@ -0,0 +1,269 @@
+import asyncio
+from motor import motor_asyncio
+from typing import Any, TypedDict
+
+import nextcord
+from nextcord import Interaction, SlashOption
+from nextcord.ext import commands, application_checks, tasks
+
+from util.nira import NIRA
+
+class AutoEmojiSetting(TypedDict):
+    """
+    `AutoEmoji.autoemoji_list`に使用される、AutoEmojiの設定を表す型です。
+    """
+    channel_id: int
+    emojis: list[str]
+
+class AutoEmoji(commands.Cog):
+    """
+    AutoEmojiは、設定チャンネルにメッセージが投稿された際に、自動で絵文字リアクションを行う機能です。
+    これを行うことでチャンネル運用がもしかしたら便利になるかもしれません。
+    """
+    def __init__(self, bot: NIRA, **kwargs: Any):
+        self.bot = bot
+        self.autoemoji_collection: motor_asyncio.AsyncIOMotorCollection = self.bot.database["autoemoji_setting"]
+        self.autoemoji_list: list[AutoEmojiSetting] = []
+        self.load_autoemoji_settings.start()
+
+    @tasks.loop(hours=1.0)
+    async def load_autoemoji_settings(self):
+        """
+        AutoEmojiの設定を、データベースからローカルへロードします。
+        """
+        self.autoemoji_list = []
+        async for autoemoji in self.autoemoji_collection.find():
+            self.autoemoji_list.append(autoemoji)
+
+    @nextcord.slash_command(
+        name="autoemoji",
+        description="Configure autoemoji settings",
+    )
+    async def autoemoji_slash(self, interaction: Interaction):
+        pass
+
+    @application_checks.guild_only()
+    @autoemoji_slash.subcommand(
+        name="add",
+        description="Add a new autoemoji",
+        description_localizations={
+            nextcord.Locale.ja: "新しい自動絵文字を追加します。",
+        }
+    )
+    async def add_autoemoji_slash(
+        self,
+        interaction: Interaction,
+        emojis: str = SlashOption(
+            name="emojis",
+            name_localizations={
+                nextcord.Locale.ja: "絵文字",
+            },
+            description="Comma separated emojis to add",
+            description_localizations={
+                nextcord.Locale.ja: "カンマ区切りで絵文字を指定してください。",
+            },
+            required=True,
+        )
+    ):
+        assert isinstance(interaction.guild, nextcord.Guild)
```

Comment:
`interaction.guild` は `Guild | None` だから他と合わせてほしい
```suggestion
        assert interaction.guild is not None
```